pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MergeRollupMinionClusterIntegrationTest.java

```diff
@@ -0,0 +1,565 @@
+    JsonNode expectedJson = postSqlQuery(sqlQuery, _brokerBaseApiUrl);
+    int[] expectedNumSegmentsQueried = {16, 7, 3};
+    int[] expectedNumMergedSegments = {2, 4, 5};
+    long expectedWatermark = 16050 * 86_400_000L;
+    String offlineTableName = TableNameBuilder.OFFLINE.tableNameWithType(SINGLE_LEVEL_ROLLUP_TEST_TABLE);
+    int numTasks = 0;
+    for (String tasks = _taskManager.scheduleTasks(Collections.singletonList(offlineTableName), true)
+        .get(MinionConstants.MergeRollupTask.TASK_TYPE); tasks != null; tasks =
+        _taskManager.scheduleTasks(Collections.singletonList(offlineTableName), true)
+            .get(MinionConstants.MergeRollupTask.TASK_TYPE), numTasks++) {
+      Assert.assertEquals(_helixTaskResourceManager.getTaskConfigs(tasks).size(), 1);
+      Assert.assertTrue(_helixTaskResourceManager.getTaskQueues()
+          .contains(PinotHelixTaskResourceManager.getHelixJobQueueName(MinionConstants.MergeRollupTask.TASK_TYPE)));
+      // Will not schedule task if there's incomplete task
+      Assert.assertNull(_taskManager.scheduleTasks(Collections.singletonList(offlineTableName), true)
+          .get(MinionConstants.RealtimeToOfflineSegmentsTask.TASK_TYPE));
+      waitForTaskToComplete();
+
+      // Check watermark
+      MergeRollupTaskMetadata minionTaskMetadata = MergeRollupTaskMetadata.fromZNRecord(
+          _taskManager.getClusterInfoAccessor().getMinionMergeRollupTaskZNRecord(offlineTableName));
+      assertNotNull(minionTaskMetadata);
+      Assert.assertEquals((long) minionTaskMetadata.getWatermarkMap().get("150days"), expectedWatermark);
+      expectedWatermark += 150 * 86_400_000L;
+
+      // Check num total doc of merged segments are the same as original segment
+      int numMergedSegments = 0;
+      for (OfflineSegmentZKMetadata metadata : _pinotHelixResourceManager.getOfflineSegmentMetadata(offlineTableName)) {
+        if (metadata.getSegmentName().startsWith("merged")) {
+          // Check merged segment zk metadata
+          assertNotNull(metadata.getCustomMap());
+          assertEquals("150days",
+              metadata.getCustomMap().get(MinionConstants.MergeRollupTask.SEGMENT_ZK_METADATA_MERGE_LEVEL_KEY));
+          // Check merged segments are time partitioned
+          assertEquals(metadata.getEndTimeMs() / (86_400_000L * 150), metadata.getStartTimeMs() / (86_400_000L * 150));
+          numMergedSegments++;
+        }
+      }
+      assertEquals(numMergedSegments, expectedNumMergedSegments[numTasks]);
+
+      // Check query response and routing
+      JsonNode actualJson = postSqlQuery(sqlQuery, _brokerBaseApiUrl);
+      int numSegmentsQueried = actualJson.get("numSegmentsQueried").asInt();
+      assertEquals(numSegmentsQueried, expectedNumSegmentsQueried[numTasks]);
+      assertTrue(actualJson.get("resultTable").get("rows").get(0).get(0).asInt() < expectedJson.get("resultTable")
```

Comment:
Can you test the exact expected result instead of `<`

Reply:
Yes, added a test to check the exact number when all the tasks are finished.